File: server/src/main/java/com/fluffy/exam/api/ExamStatusConverter.java
package com.fluffy.exam.api;

import com.fluffy.exam.domain.ExamStatus;
import org.springframework.core.convert.converter.Converter;
import org.springframework.lang.NonNull;
import org.springframework.stereotype.Component;

@Component
public class ExamStatusConverter implements Converter<String, ExamStatus> {

    @Override
    public ExamStatus convert(@NonNull String source) {
        return ExamStatus.from(source);
    }
}

File: server/src/main/kotlin/com/fluffy/exam/ui/ExamStatusConverter.kt
package com.fluffy.exam.ui

import com.fluffy.exam.domain.ExamStatus
import org.springframework.core.convert.converter.Converter
import org.springframework.stereotype.Component

@Component
class ExamStatusConverter : Converter<String, ExamStatus> {

    override fun convert(source: String): ExamStatus {
        return ExamStatus.from(source)
    }
}
